dynamax/linear_gaussian_ssm/parallel_inference_test.py

```diff
@@ -12,86 +12,108 @@
+        dynamics_weights=F,
+        dynamics_input_weights=B,
+        dynamics_bias=b,
+        dynamics_covariance=Q,
+        emission_weights=H,
+        emission_input_weights=D,
+        emission_bias=d,
+        emission_covariance=R,
+    )
     return params, lgssm
-        
 
-def make_dynamic_lgssm_params(num_timesteps, latent_dim=4, observation_dim=2, seed=0):
-    key = jr.PRNGKey(seed)
-    key, key_f, key_r, key_init = jr.split(key, 4)
+
+def make_dynamic_lgssm_params(num_timesteps):
+    latent_dim = 4
+    observation_dim = 2
+    input_dim = 3
+
+    keys = jr.split(jr.PRNGKey(1), 9)
 
     dt = 0.1
-    f_scale = jr.normal(key_f, (num_timesteps,)) * 0.5
-    F = f_scale[:,None,None] * jnp.tile(jnp.eye(latent_dim), (num_timesteps, 1, 1))
-    F += dt * jnp.eye(latent_dim, k=2)
-
-    Q = 1. * jnp.kron(jnp.array([[dt**3/3, dt**2/2],
-                                 [dt**2/2, dt]]),
-                      jnp.eye(latent_dim // 2))
-    assert Q.shape[-1] == latent_dim
-    H = jnp.eye(observation_dim, latent_dim)
-
-    r_scale = jr.normal(key_r, (num_timesteps,)) * 0.1
-    R = (r_scale**2)[:,None,None] * jnp.tile(jnp.eye(observation_dim), (num_timesteps, 1, 1))
-    
-    μ0 = jnp.array([0.,0.,1.,-1.])
+
+    F = jnp.eye(4)[None] + dt * jnp.eye(4, k=2)[None] + 0.1 * jr.normal(keys[0], (num_timesteps, latent_dim, latent_dim))
+    B = 0.2 * jr.normal(keys[4], (num_timesteps, latent_dim, input_dim))
+    b = 0.2 * jr.normal(keys[6], (num_timesteps, latent_dim))
+    q_scale = jr.normal(keys[1], (num_timesteps, 1, 1)) ** 2
+    Q = q_scale * jnp.kron(jnp.array([[dt**3 / 3, dt**2 / 2], [dt**2 / 2, dt]]), jnp.eye(2))[None]
+
+    H = jnp.eye(2, 4)[None] * 0.1 * jr.normal(keys[2], (num_timesteps, observation_dim, latent_dim))
+    D = 0.2 * jr.normal(keys[5], (num_timesteps, observation_dim, input_dim))
+    d = 0.2 * jr.normal(keys[7], (num_timesteps, observation_dim))
+    r_scale = jr.normal(keys[3], (num_timesteps, 1, 1)) ** 2
+    R = r_scale * jnp.eye(2)[None]
+
+    μ0 = jnp.array([1.0, -2.0, 1.0, -1.0])
     Σ0 = jnp.eye(latent_dim)
 
-    lgssm = LinearGaussianSSM(latent_dim, observation_dim)
-    params, _ = lgssm.initialize(key_init,
-                                initial_mean=μ0,
-                                initial_covariance=Σ0,
-                                dynamics_weights=F,
-                                dynamics_covariance=Q,
-                                emission_weights=H,
-                                emission_covariance=R)
+    lgssm = LinearGaussianSSM(latent_dim, observation_dim, input_dim)
+    params, _ = lgssm.initialize(
+        keys[8],
+        initial_mean=μ0,
+        initial_covariance=Σ0,
+        dynamics_weights=F,
+        dynamics_input_weights=B,
+        dynamics_bias=b,
+        dynamics_covariance=Q,
+        emission_weights=H,
+        emission_input_weights=D,
+        emission_bias=d,
+        emission_covariance=R,
+    )
     return params, lgssm
```

Comment:
The original synthetic data for testing the time-varying case was too simplistic to capture some bugs while I was developing the code. So I updated the test case which has more time variation of parameters.